Update the CC310 IAR libraries

RonEld opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->
After testing the Mbed TLS On Target Tests with IAR, on NRF52840 platform, it was discovered that the Cryptocell library is not initiated correct. This fix replaces the libraries received  by Mail, with libraries built from the official Cryptocell 310 release, version 1.1.0.1285, using IAR version 7.80.1.11864 and the `NRF52840` platform registers base address.

Tested with GreenTea Mbed TLS On Target Tests.



### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

Commit:
Update the CC310 IAR libraries

Update the cryptocell 310 IAR libraries, since the previous ones were not
built correct. The libraries were built from the Cryptocell 310
release version 1.1.0.1285, using IAR version 7.80.1.11864.

0xc0170 commented:
Can this be captured in the readme for these libs (version of the code used, toolchain version). Same would be good to have inthe commit msg as well

RonEld commented:
@0xc0170 
>Can this be captured in the readme for these libs (version of the code used, toolchain version). Same would be good to have inthe commit msg as well

In this case, I will update this PR with the ARM and GCC_ARM libraries as well, to be synced

ciarmcom commented:
@RonEld, thank you for your changes.
@ARMmbed/mbed-os-crypto @ARMmbed/mbed-os-maintainers please review.

RonEld commented:
I am currently working on updating the other toolchain libraries for specific version information, and updating the readme file, as requested earlier.

yanesca reviewed (commented):
The changes are only binaries, there is nothing to review.

cmonr commented:
Waiting on https://github.com/ARMmbed/mbed-os/pull/9131 for a fix to Travis CI.

Commit:
Update the CC310 ARM libraries

Update the cryptocell 310 ARM libraries with known version.
The libraries were built from the Cryptocell 310 release version
1.1.0.1285, using `ARM Compiler 5.06 update 4 (build 422)`.

Commit:
Update the CC310 readme file

Update the Cryptocell 310 readme file with the binary library version
information.

RonEld commented:
I have amended the commit message with version information, and uploaded the libraries for ARM and GCC_ARM with the specific version information.
in addition, I have updated the `Readme.md` file with the version information. Should we ask someone from the docs team to review the changes in the readme?

0xc0170 commented on features/cryptocell/FEATURE_CRYPTOCELL310/Readme.md:
```diff
@@ -8,6 +8,13 @@ The CC 310 driver consists of three libraries:
+* The CC 310 libraries were built from version `arm_sw-cc310-1.1.0.1285`
+* The `IAR` libraries were built using `IAR ANSI C/C++ Compiler V7.80.1.11864/W32 for ARM` with `--cpu Cortex-M4f`.
+* The `ARM` libraries were built using `ARM Compiler 5.06 update 4 (build 422)` with `--cpu cortex-m4`.
+* The `GCC_ARM` libraries were built using `arm-none-eabi-gcc 7.3.1 20180622 (release)` with `-mcpu=cortex-m4`.
```
tested with GCC ARM 6 as that one is supported, this would rather be recompiled

Commit:
Update the CC310 GCC_ARM libraries

Update the cryptocell 310 GCC_ARM libraries with known version.
The libraries were built from the Cryptocell 310 release
version 1.1.0.1285, using arm-none-eabi-gcc 6.3.1 20170620 (release) (release).

RonEld commented:
1. I removed the previous commit with the build with gcc 7.3.1
2. I amended https://github.com/ARMmbed/mbed-os/pull/9122/commits/244647088eaad2bed4e22ffbd3cbf9fd2dd10782 with the new version
3. I rebuilt the library with gcc 6.3.1, and uploaded it in a new commit

0xc0170 reviewed (approved):
One remaining question: increase binary sizes (in case of IAR 240 percent reported by Github - is that correct) ?

You can check by viewing files in the PR, Github reports the size differences

RonEld commented:
@0xc0170 
>One remaining question: increase binary sizes (in case of IAR 240 percent reported by Github - is that correct) ?

Since the previous IAR libraries were actually broken. I assume this is correct. I haven't built them, and they were built with wchar defined as short ( Probably for IAR 8 ). This could explain why it was broken, and why the size was smaller.

RonEld commented:
@0xc0170 
>Should we ask someone from the docs team to review the changes in the readme?

0xc0170 commented:
@AnotherButler for later review, she is currently OoO. Reviewed by a maintainer should be sufficient for this change
